src/workarounds/forwarddiff_rules.jl

```diff
@@ -180,15 +180,31 @@ function self_consistent_field(basis_dual::PlaneWaveBasis{T};
 
     ## Implicit differentiation
     hamψ_dual = ham_dual * ψ_dual
-    δHψ = [ForwardDiff.partials.(δHψₖ, 1) for δHψₖ in hamψ_dual]
-    δψ, response = solve_ΩplusK(basis, ψ, -δHψ, occupation;
-                                tol_cg=scfres.norm_Δρ, verbose=response.verbose)
-    δρ  = compute_δρ(basis, ψ, δψ, occupation)
+    N = ForwardDiff.npartials(T)
+    δresults = ntuple(N) do j
```

Comment:
would be good to use more distinctive indices, since this can get confusing. How about Npartials and \alpha?